misc/docerina/src/main/java/org/ballerinalang/docgen/docs/BallerinaDocGenerator.java

```diff
@@ -226,23 +262,98 @@ private static void writeAPIDocs(ModuleLibrary moduleLib, Path output, boolean i
     }
 
     private static void copyDocerinaUI(Path output) {
-        File source = Path.of(System.getProperty("ballerina.home"), "lib", "tools", "doc-ui").toFile();
-        File dest;
-        if (source.exists()) {
-            dest = output.toFile();
-            try {
-                FileUtils.copyDirectory(source, dest);
-            } catch (IOException e) {
-                log.error("Failed to copy the doc ui.", e);
+        String source = RepoUtils.getRemoteRepoURL();
+        Path docsDirPath = ProjectUtils.createAndGetHomeReposPath().resolve(DOCS_FOLDER_NAME);
+        Path sha256FilePath = docsDirPath.resolve(SHA256_HASH_FILE_NAME);
+        Path zipFilePath = docsDirPath.resolve(BALLERINA_DOC_UI_ZIP_FILE_NAME);
+        OkHttpClient client = new OkHttpClient();
+        Request request = new Request.Builder()
+                .get()
+                .url(source)
+                .build();
+        try (Response response = client.newCall(request).execute()) {
+            if (response.code() == HTTP_OK && response.body() != null) {
+                ResponseBody responseBody = response.body();
+                JsonObject jsonResponse = JsonParser.parseReader(responseBody.charStream()).getAsJsonObject();
+                String sha256HashValue = jsonResponse.get(JSON_KEY_HASH_VALUE).getAsString();
+                String zipFileURL = jsonResponse.get(JSON_KEY_FILE_URL).getAsString();
+                if (!Files.exists(sha256FilePath) || !Files.exists(zipFilePath)) {
+                    if (!docsDirPath.toFile().exists()) {
+                        Files.createDirectories(docsDirPath);
+                    }
+                    writeFileInCache(zipFileURL, sha256HashValue, zipFilePath, sha256FilePath);
+                } else {
+                    String hashValueInCache = Files.readString(sha256FilePath).trim();
+                    if (!sha256HashValue.equals(hashValueInCache)) {
+                        writeFileInCache(zipFileURL, sha256HashValue, zipFilePath, sha256FilePath);
+                    }
+                }
+                copyDocUIToProjectDir(output, zipFilePath);
+            } else {
+                throw new IOException("Response failed with status code: " + response.code());
             }
-        } else {
-            dest = output.resolve("index.html").toFile();
-            try {
-                FileUtils.copyInputStreamToFile(BallerinaDocGenerator.class
-                        .getResourceAsStream("/doc-ui/index.html"), dest);
-            } catch (IOException e) {
-                log.error("Failed to copy the doc ui.", e);
+        } catch (IOException e) {
+            if (Files.exists(zipFilePath)) {
+                String warning = """
+                        WARNING: Unable to fetch the latest UI from the central.
```

Comment:
Since we are already saying that it failed to fetch API doc UI in the `writeInCache`, shall we remove the `Unable to fetch ...` part from the warning log?

Reply:
It wouldn't be printed unless the code returns an error, so it's better to include that in the warning message.